<launch>


     <!-- run ROSAria to control the pioneer -->
     <node pkg="rosaria" type="RosAria" name="Pioneer3AT_rosaria" output="screen">
          <rosparam file="$(find pioneer3at)/config/rosaria/rosaria.yaml" command="load" />

          <remap from="~pose"            to="/Pioneer3AT/pose" />
          <remap from="~sonar"           to="/Pioneer3AT/sonar" />
          <remap from="~cmd_vel"         to="/Pioneer3AT/cmd_vel" />
          <remap from="~bumper_state"    to="/Pioneer3AT/bumper_state" />
          <remap from="~battery_voltage" to="/Pioneer3AT/battery_voltage" />
     </node>

     <!-- SICK -->

     <node pkg="sicktoolbox_wrapper" type="sicklms" name="Pioneer3AT_sicklms" >
          <rosparam file="$(find pioneer3at)/config/sick_wrapper/LMS_200.yaml" command="load" />
          <remap from="/scan" to="/Pioneer3AT/laserscan" />
     </node>

     <!-- publish static transform from the center of the robot to the laser rangefinder (required for slam_gmapping) -->
     <node pkg="tf" type="static_transform_publisher" name="Pioneer3AT_laserscan_tf" args="0.17 0 0.4 0 0 0 Pioneer3AT/base_link Pioneer3AT/laserscan 33" />

     <param name="robot_description" command="$(find xacro)/xacro.py $(find pioneer3at)/urdf/pioneer3at.xacro" />

     <node pkg="robot_state_publisher" type="state_publisher" name="Pioneer3AT_tf_broadcaster">
          <!-- <param name="tf_prefix" type="string" value="Pioneer3AT"/> -->
          <!--<param name="tf_prefix" type="string" value="Pioneer3AT"/>-->
          <param name="publish_frequency" type="double" value="20.0"/>

          <!--<remap from="robot_description" to="/Pioneer3AT/urdf" />-->
          <remap from="joint_states" to="Pioneer3AT/joint_states"/>
     </node>

     <node pkg="pioneer3at" type="joint_state_publisher" name="Pioneer3AT_joints_publisher">
          <param name="tf_future_date" type="double" value="0.05" />

          <remap from="joint_states" to="Pioneer3AT/joint_states"/>
     </node>

     <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager"/>
     <node pkg="nodelet" type="nodelet" name="cmd_vel_mux" args="load yocs_cmd_vel_mux/CmdVelMuxNodelet nodelet_manager">
          <remap from="/cmd_vel_mux/output/cmd_vel"           to="/Pioneer3AT/cmd_vel" />

          <param name="yaml_cfg_file" value="$(find pioneer3at)/config/cmd_vel_mux.yaml"/>
     </node>

     <!--<node pkg="gmapping" type="slam_gmapping" name="Pioneer3AT_gmapping" output="screen">
          <rosparam file="$(find pioneer3at)/config/gmapping/gmapping.yaml" command="load" />
          <param name="occ_map_frame_" type="str" value="occ_map" />
          <remap from="scan"          to="/Pioneer3AT/laserscan"/>
          <remap from="map"           to="/Pioneer3AT/map"/>
          <remap from="map_metadata"  to="/Pioneer3AT/map_metadata"/>
          <remap from="occ_map"       to="/Pioneer3AT/occ_map" />
     </node>-->

       <node pkg="gmapping" type="slam_gmapping" name="Pioneer3AT_gmapping">

            <rosparam file="$(find pioneer3at)/config/gmapping/gmapping_multi.yaml" command="load" />
            <param name="base_frame" type="str" value="/Pioneer3AT/base_link" />
            <param name="map_frame" type="str" value="/Pioneer3AT/map" />
            <param name="odom_frame" type="str" value="/Pioneer3AT/odom" />

            <param name="occ_map_frame_" type="str" value="occ_map" />
            <remap from="scan"          to="/Pioneer3AT/laserscan"/>
            <remap from="map"           to="/Pioneer3AT/map"/>
            <remap from="map_metadata"  to="/Pioneer3AT/map_metadata"/>
            <remap from="occ_map"       to="/Pioneer3AT/occ_map" />
        </node>


     <node pkg="move_base" type="move_base" name="Pioneer3AT_move_base">
          <rosparam file="$(find pioneer3at)/config/move_base/move_base.yaml" command="load"/>
          <rosparam file="$(find pioneer3at)/config/move_base/costmap_common.yaml" command="load" ns="global_costmap" />
          <rosparam file="$(find pioneer3at)/config/move_base/costmap_common.yaml" command="load" ns="local_costmap" />
          <rosparam file="$(find pioneer3at)/config/move_base/costmap_global.yaml" command="load"  />
          <rosparam file="$(find pioneer3at)/config/move_base/costmap_local.yaml" command="load"/>

          <remap from="cmd_vel"               to="/Pioneer3AT/move_base/cmd_vel"  />
          <remap from="move_base/cancel"      to="/Pioneer3AT/move_base/cancel"   />
          <remap from="move_base/feedback"    to="/Pioneer3AT/move_base/feedback" />
          <remap from="move_base/goal"        to="/Pioneer3AT/move_base/goal" />
          <remap from="move_base/result"      to="/Pioneer3AT/move_base/result" />
          <remap from="move_base/status"      to="/Pioneer3AT/move_base/status" />
          <remap from="move_base_simple/goal" to="/Pioneer3AT/move_base_simple/goal" />

     </node>

     <node pkg="rqt_robot_steering" type="rqt_robot_steering" name="Pioneer3AT_rqt_teleop">
          <param name="default_topic" type="str" value="/Pioneer3AT/rqt/cmd_vel" />
          <param name="default_vx_max" type="double" value="0.8" />
          <param name="default_vx_min" type="double" value="-0.8" />
          <param name="default_vw_max" type="double" value="1.5" />
          <param name="default_vw_min" type="double" value="-1.5" />
     </node>

      <!--<node pkg="pioneer3at" type="map_node_real" name="Pioneer3AT_map_node">
          <param name="map_topic" type="str" value="/Pioneer3AT/map" />
          <param name="gmapping_map_topic" type="str" value="/Pioneer3AT/gmapping_map" />
          <param name="occ_map_topic" type="str" value="/Pioneer3AT/occ_map"/>
          <param name="occ_gmapping_map_topic" type="str" value="/Pioneer3AT/occ_gmapping_map"/>
          <param name="other_map_topic" type="str" value="/robot2/occ_map" />
          <param name="robots_pose_topic" type="str" value="/Pioneer3AT/robots_poses" />
          <param name="other_robot_pose_topic" type="str" value="/robot2/robots_poses" />
          <param name="signal_topic" type="str" value="/Pioneer3AT/signal_topic" />
      </node>-->

       <node pkg="pioneer3at" type="new_explorer_real_ent" name="Pioneer3AT_window_explorer" required="true">
            <param name="robot_topic" type="str" value="Pioneer3AT" />
            <param name="goal_topic" type="str" value="/Pioneer3AT/move_base_simple/goal" />
            <param name="map_topic" type="str" value="/Pioneer3AT/map" />
            <param name="occ_map_topic" type="str" value="/Pioneer3AT/occ_map" />
            <param name="pose_topic" type="str" value="/Pioneer3AT/pose" />
            <param name="base_link_topic" type="str" value="/Pioneer3AT/base_link" />
            <param name="map_merge_topic" type="str" value="/Pioneer3AT/map_merge_topic" />
            <param name="other_robot_pose_topic" type="str" value="/robot2/odom" />
            <param name="robots_poses_topic" type="str" value="/Pioneer3AT/robots_poses" />
            <param name="signal_topic" type="str" value="/Pioneer3AT/signal_topic" />
            <param name="goal_status_topic" type="str" value="/Pioneer3AT/move_base/status" />
            <param name="odom_topic" type="str" value="/Pioneer3AT/odom" />
        </node>

     <node pkg="rviz" type="rviz" name="Pioneer3AT_rviz" args="-d $(find pioneer3at)/config/rviz/settings.rviz"/>

        <!--<node pkg="usb_cam" type="usb_cam_node" name="usb_cam_node">
                <param name="video_device" type="string" value="/dev/video1"/>

                <param name="pixel_format" type="string" value="mjpeg" />
                <param name="io_method" type="string" value="mmap" />

                <param name="image_width" type="int" value="640" />
                <param name="image_height" type="int" value="480" />

                <param name="camera_frame_id" type="string" value="/Pioneer3AT/usb_cam_frame" />
                <param name="camera_info_url" type="string" value="file://$(find usb_cam)/cal.yml" />
        </node>-->

       <!--<node pkg="image_view" type="video_recorder" name="video_recorder">
            <param name="filename" type="string" value="/home/rcolares/catkin_ws/src/ros-pioneer3at/maps/pioneer.avi"/>
            <param name="fps" type="int" value="30"/>

            <remap from="image" to="/usb_cam_node/image_raw"/>
       </node>-->


</launch>








